supervision/detection/core.py

```diff
@@ -977,15 +983,21 @@ def empty(cls) -> Detections:
             xyxy=np.empty((0, 4), dtype=np.float32),
             confidence=np.array([], dtype=np.float32),
             class_id=np.array([], dtype=int),
+            metadata=metadata if metadata is not None else {},
         )
 
     def is_empty(self) -> bool:
         """
         Returns `True` if the `Detections` object is considered empty.
         """
-        empty_detections = Detections.empty()
-        empty_detections.data = self.data
-        return self == empty_detections
+        return (
+            len(self.xyxy) == 0
+            and (self.mask is None or len(self.mask) == 0)
+            and (self.class_id is None or len(self.class_id) == 0)
+            and (self.confidence is None or len(self.confidence) == 0)
+            and (self.tracker_id is None or len(self.tracker_id) == 0)
+            and not self.data
+        )
```

Comment:
This widens the description of what `empty` is allowed to be. Let's keep it as it was. It'll get simplified when we solve #943

Reply:
I like the optional metadata argument for `empty`, but let's avoid checking for it here (similarly to how we avoided `data` check).

Suppose there is a model that returns values `{detection, X, Y}` when detections are successful and `{[], X}` when nothing is detected. This prevents the developers from using the metadata system if they ever want to add `Y`.

Ultimately, I'd like to simplify this to `len(self.xyxy) == 0`, but only after the next release, to have time for testing.